lars-reimann/api-editor
# feat: Apply move

Closes #299

### Summary of Changes

- Add a MoveAnnotationProcessor
- Apply MoveAnnotations during infer
- Add a CleanupModuleProcessor which removes empty modules after all Annotations have been applied
- Add new test cases for 
- update test cases (some tests had conflicting qualifiedNames or added the package name to the qualified name. those errors should be fixed now)

### Testing instructions

1. Mark declarations on different levels with the @move annotation
2. Check that code generation for stub files and adapters work (In the data folder is a "minimal_test_package.json", which is convenient for quick testing because of its smaller size compared to the entire scikit library.)
3. Check unit tests
